apps/api/src/app/plugins/bffCache.ts

```diff
@@ -39,32 +42,36 @@ export const bffCache: FastifyPluginCallback<BffCacheOptions> = (fastify, opts,
     return
   })
 
-  fastify.addHook('onSend', function (req, reply, payload, next) {
-    const isCacheHit = reply.getHeader(HEADER_NAME) === 'HIT'
-    const cacheTtl: number | undefined = getTtlFromResponse(reply, ttl)
-    const isStatus200 = reply.statusCode >= 200 && reply.statusCode < 300
+  fastify.addHook('onSend', async function (req, reply, payload) {
+    const isCacheHit = reply.getHeader(HEADER_NAME) === 'HIT';
+    const cacheTtl: number | undefined = getTtlFromResponse(reply, ttl);
+    const isStatus200 = reply.statusCode >= 200 && reply.statusCode < 300;
 
     // If the cache is a hit, or is non-cacheable, we just proceed with the request
     if (isCacheHit || cacheTtl === undefined || !isStatus200) {
-      next();
       return;
     }
 
     // If there is no cached data, then its a cache-miss
-    reply.header(HEADER_NAME, 'MISS')
 
-    let key = getKey(req)
-    setCache(key, payload, cacheTtl, fastify).catch(e => {
-      fastify.log.error(`Error setting key ${key} from cache`, e)
-      return null
-    })
-    reply.header(CACHE_CONTROL_HEADER, getCacheControlHeaderValue(cacheTtl))
-    next()
-  })
+    let contents = '';
+    for await (const chunk of payload as ReadableStream) {
+      contents += chunk.toString(); // Process each chunk of data
+    }
```

Comment:
This is the fix: consume the payload stream once, so it can be reused in the response and stored in the cache.